api/repositories/service_offering_repository_test.go

```diff
@@ -317,4 +330,136 @@ var _ = Describe("ServiceOfferingRepo", func() {
 			})
 		})
 	})
+
+	Describe("Delete", func() {
+		var (
+			plan      *korifiv1alpha1.CFServicePlan
+			offering  *korifiv1alpha1.CFServiceOffering
+			instance  *korifiv1alpha1.CFServiceInstance
+			binding   *korifiv1alpha1.CFServiceBinding
+			message   repositories.DeleteServiceOfferingMessage
+			deleteErr error
+		)
+
+		BeforeEach(func() {
+			offering = &korifiv1alpha1.CFServiceOffering{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: rootNamespace,
+					Name:      uuid.NewString(),
+				},
+				Spec: korifiv1alpha1.CFServiceOfferingSpec{
+					ServiceOffering: services.ServiceOffering{
+						Name:        "my-offering",
+						Description: "my offering description",
+						Tags:        []string{"t1"},
+						Requires:    []string{"r1"},
+					},
+				},
+			}
+			Expect(k8sClient.Create(ctx, offering)).To(Succeed())
+
+			plan = &korifiv1alpha1.CFServicePlan{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: rootNamespace,
+					Name:      uuid.NewString(),
+					Labels: map[string]string{
+						korifiv1alpha1.RelServiceOfferingGUIDLabel: offering.Name,
+					},
+				},
+				Spec: korifiv1alpha1.CFServicePlanSpec{
+					ServicePlan: services.ServicePlan{
+						Name:        "my-service-plan",
+						Free:        true,
+						Description: "service plan description",
+					},
+					Visibility: korifiv1alpha1.ServicePlanVisibility{
+						Type: korifiv1alpha1.PublicServicePlanVisibilityType,
+					},
+				},
+			}
+			Expect(k8sClient.Create(ctx, plan)).To(Succeed())
+
+			instance = createServiceInstanceCR(ctx, k8sClient, uuid.NewString(), space.Name, "my-service-instance", "secret-name")
+			instance.Spec.PlanGUID = plan.Name
+			instance.Finalizers = append(instance.Finalizers, korifiv1alpha1.CFManagedServiceInstanceFinalizerName)
+
+			Expect(k8sClient.Update(ctx, instance)).To(Succeed())
+
+			binding = &korifiv1alpha1.CFServiceBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      uuid.NewString(),
+					Namespace: space.Name,
+					Labels: map[string]string{
+						korifiv1alpha1.PlanGUIDLabelKey: plan.Name,
+					},
+				},
+				Spec: korifiv1alpha1.CFServiceBindingSpec{
+					Service: corev1.ObjectReference{
+						Kind:       "CFServiceInstance",
+						APIVersion: korifiv1alpha1.SchemeGroupVersion.Identifier(),
+						Name:       instance.Name,
+					},
+					AppRef: corev1.LocalObjectReference{
+						Name: "some-app-guid",
+					},
+				},
+			}
+
+			binding.Finalizers = append(binding.Finalizers, korifiv1alpha1.CFServiceBindingFinalizerName)
+			Expect(k8sClient.Create(ctx, binding)).To(Succeed())
+
+			message = repositories.DeleteServiceOfferingMessage{GUID: offering.Name}
+		})
+
+		JustBeforeEach(func() {
+			createRoleBinding(ctx, userName, spaceDeveloperRole.Name, space.Name)
```

Comment:
Generally, `JustBeforeEach` should be only used to invoke the code under test. Therefore, creating the role binding should be done in the related `BeforeEach`